src/character.cpp

```diff
@@ -227,6 +224,23 @@ std::pair<int, int> Character::get_best_sight( const item &gun, double recoil )
     return std::make_pair( sight_speed_modifier, limit );
 }
 
+int Character::get_most_accurate_sight( const item &gun ) const
+{
+    if( !gun.is_gun() ) {
+        return 0;
+    }
+
+    int limit = effective_dispersion( gun.type->gun->sight_dispersion );
+    for( const auto e : gun.gunmods() ) {
+        const islot_gunmod &mod = *e->type->gunmod;
+        if( mod.aim_speed >= 0 ) {
```

Comment:
The function above (which somewhat reminds me this one, but stripped of `sight_speed_modifier`) also checks if `mod.sight_dispersion < 0` and skips the `mod` if it's true. Can `limit` become negative here, will it still be meaningful in this case?

Reply:
I can add that check, but item factory guarantees it anyway:
```
if( ( type->gunmod->sight_dispersion < 0 ) != ( type->gunmod->aim_speed < 0 ) ){
    msg << "gunmod must have both sight_dispersion and aim_speed set or neither of them set" << "\n";
}
```